WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListViewModel.swift

```diff
@@ -0,0 +1,239 @@
+import SwiftUI
+
+final class BlogListViewModel: NSObject, ObservableObject {
```

Comment:
While reviewing this file, I though of a suggestion that would simplify the code a bit. I pushed my proposal to `hack-week_site-picker-redesign-hassaan-suggestion-2` [GH Link](https://github.com/wordpress-mobile/WordPress-iOS/compare/hack-week_site-picker-redesign...hack-week_site-picker-redesign-hassaan-suggestion-2?expand=1). The idea is to remove the dependency on CoreData for the remaining sites section, and calculate it using (Remaining = All Blogs - Pinned - Recents). Let me know what you think.

Reply:
Are the sites still sorted alphabetically this way? If yes, then we can go ahead with it. I'll have to test it though as there are bunch of cases.

Reply:
Is the PR nesting on an out of date version of this branch? Because the search and edit mode bug appears to be back on the suggestion-2 branch.

Reply:
It should be based on the latest version. I can take a look at how to fix that issue later tonight, unless you beat me to it. Can you share a link or a description of the issue though?

Reply:
> he search and edit mode bug appears to be back on the suggestion-2 branch.

I just pushed a commit to the suggestion branch resolving this issue a5fee1f23bdba3cd39ec70266904784ceac1e244

Reply:
> Are the sites still sorted alphabetically this way?

I believe right now they're not, but we can change the sort descriptor of `allBlogsController` to sort using the name, if we are okay with sorting the search results alphabetically as well.

Reply:
> if we are okay with sorting the search results alphabetically as well.

That should be fine.

Reply:
>if we are okay with sorting the search results alphabetically as well.
> ...
>That should be fine.

Done in 2db1999fb48f05d978a1dd9623742a5308e3dd8f 👍

Reply:
I have tested the changes on the suggestion-2 branch and everything is looking fine. I think we can merge those in here, wait for the CI and merge this to trunk. Wdyt?

Reply:
I'll do a PR once we know whether to do full-rollout or not to do a clean up. If we don't rollout, either delete the new code or update it, if we do, then delete the legacy and introduce the Coordinator that aims to resolve those issues.